workspaces/adventure-pack/goodies/typescript/Iterator.from/index.ts

```diff
@@ -0,0 +1,43 @@
+import "../Iterator.prototype.toIterable";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  class Iterator<T> {
+    static from<TElem>(
+      object:
+        | Iterator<TElem>
+        | Iterable<TElem>
+        | { next(): IteratorResult<TElem> },
+    ): IterableIterator<TElem>;
+  }
+}
+
+(globalThis as Record<string, unknown>).Iterator ??= {};
+
+(
+  (globalThis as Record<string, unknown>).Iterator as Record<string, unknown>
+).from ??= function <T>(
+  object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+): IterableIterator<T> {
+  if (typeof (object as Iterable<T>)[Symbol.iterator] === "function") {
+    return (object as Iterable<T>)[Symbol.iterator]() as IterableIterator<T>;
```

Comment:
There's no guarantee that the return value of `object[Symbol.iterator]` is an `IterableIterator`, so I'm not sure that cast is correct...